lib/di.dart

```diff
@@ -497,8 +498,30 @@ Future<void> setup({
       settingsStore: getIt.get<SettingsStore>(),
       fiatConvertationStore: getIt.get<FiatConversionStore>()));
 
+  getIt.registerFactory(
+    () => ExchangeViewModel(
+      getIt.get<AppStore>(),
+      _tradesSource,
+      getIt.get<ExchangeTemplateStore>(),
+      getIt.get<TradesStore>(),
+      getIt.get<AppStore>().settingsStore,
+      getIt.get<SharedPreferences>(),
+      getIt.get<ContactListViewModel>(),
+      getIt.get<FeesViewModel>(),
+    ),
+  );
+
+  getIt.registerSingleton(
+    SwapManager(
+      tradesStore: getIt.get<TradesStore>(),
+      settingsStore: getIt.get<SettingsStore>(),
+    ),
+  );
```

Comment:
I don't see a need for this also, just go through the trades when listing them and if the trade was created in less than 24 hours, then initiate its provider and get its status and add a timer for it until its creation time exceeds 24 hours or its status is completed (`success`, `expired`, `failed`, etc....)